great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -270,12 +280,14 @@ def get_batch_list_from_batch_request(
 
         batch_list: List[Batch] = []
         column_splitter = self.column_splitter
+        batch_spec_kwargs: dict[str, str | dict | None]
```

Comment:
I am fine with this. I would also be fine with `Any` here since the values are really unstructured. That is, leave this as is for now but if value types start getting big and unwieldy because of how our system uses batch_spec_kwargs, we can switch to `Any` since the type checking probably isn't adding value.